.github/workflows/label.yml

```diff
@@ -0,0 +1,50 @@
+name: Labeler
+on:
+  pull_request_target:
+    types: closed
+
+jobs:
+  label:
+
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      issues: write
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          sparse-checkout: .github/scripts
+      - name: Add release labels on merge
+        run: |
+          PR_NUMBER="${{ github.event.pull_request.number }}"
+          
+          echo "Base REF:   $GITHUB_BASE_REF"
+          echo "PR:         https://github.com/$GITHUB_REPOSITORY/pull/$PR_NUMBER"
+          
+          if [ "$GITHUB_BASE_REF" == "main" ]; then
+            LAST_MAJOR="$(gh api /repos/$GITHUB_REPOSITORY/branches --paginate --jq .[].name | grep '^release/' | cut -d '/' -f 2 | cut -d '.' -f 1 | sort -n -r | head -n 1)"
+            NEXT_MAJOR="$(($LAST_MAJOR + 1))"
+            LABEL="release/$NEXT_MAJOR.0.0"
+            BACKPORT_LABEL="backport/main"
+          elif [[ "$GITHUB_BASE_REF" = release/* ]]; then
+            MAJOR_MINOR="$(echo $GITHUB_BASE_REF | cut -d '/' -f 2)"
+            LAST_MICRO="$(gh api /repos/$GITHUB_REPOSITORY/tags --jq .[].name | sort -n -r | grep $MAJOR_MINOR | head -n 1 | cut -d '.' -f 3)"
+            NEXT_MICRO="$(($LAST_MICRO + 1))"
+            LABEL="release/$MAJOR_MINOR.$NEXT_MICRO"
```

Comment:
Just a nit so feel free to ignore, but I am used to the `MAJOR.MINOR.PATCH` terminology. 

```suggestion
            LAST_PATCH="$(gh api /repos/$GITHUB_REPOSITORY/tags --jq .[].name | sort -n -r | grep $MAJOR_MINOR | head -n 1 | cut -d '.' -f 3)"
            NEXT_PATCH="$(($LAST_PATCH + 1))"
            LABEL="release/$MAJOR_MINOR.$NEXT_PATCH"
```